assemblyline/filestore/transport/http.py

```diff
@@ -108,3 +109,24 @@ def get(self, path):
 
     def put(self, dst_path, content):
         raise TransportException("READ ONLY TRANSPORT: Method not implemented")
+
+    def read(self, path):
+        path = self.normalize(path)
+        resp = self.session.get(path, auth=self.auth, cert=self.pki, verify=self.verify, stream=True)
+        if resp.ok:
+            return TransportReadStreamHTTP(resp)
+        else:
+            raise TransportException("[%s] %s: %s" % (resp.status_code, resp.reason, path))
+
+
+# TODO: Create an extension of the base class TransportFile
+
+class TransportReadStreamHTTP(TransportReadStream):
+    def __init__(self, response):
+        self.response = response
+
+    def close(self):
+        self.response.close()
+
+    def read(self, chunk_size=1024):
+        return next(self.response.iter_content(chunk_size=chunk_size))
```

Comment:
Correct me if I'm wrong, but we probably don't want to be created a new iterator on every call to read